src/Mvc/Mvc.Analyzers/src/TagHelpersInCodeBlocksAnalyzer.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class TagHelpersInCodeBlocksAnalyzer : DiagnosticAnalyzer
```

Comment:
@pranavkm / @rynowak This PR is in design because it's my first take at adding Anaylzers and I want to make sure I'm not doing anything catastrophically wrong design wise.